### PR TITLE
Upgrade to protobuf-gradle-plugin:0.4.0

### PR DESCRIPTION
**Pull request by zhangkun83**
Use the plugin to compile nano for golden test and get rid of the shell script.

@ejona86 please review

### CODE REVIEW

File: compiler/build.gradle [hunk 1]
```diff
@@ -73,9 +73,46 @@ model {
...
+          option 'nano=true'
+        }
+      }
+    }
+    compileClasspath += configurations.testNanoCompile
+  }
+}
+
+checkstyleTestNano {
+  source = fileTree(dir: "src/testNano", include: "**/*.java")
```
Comment (ejona86) [inline compiler/build.gradle]: What does this do? I don't see any src/testNano.

Comment (zhangkun83) [inline compiler/build.gradle]: checkstyle creates a checkstyleTestNano for the testNano sourceSet, which seems to include everything under the project, including ``build/generated-sources``. I just need to point the source to nothing, but I think it's nice to be consistent with other checkstyle tasks.

File: compiler/build.gradle [hunk 2]
```diff
@@ -73,9 +73,46 @@ model {
...
+          option 'ignore_services=true'
+        }
+      }
+      plugins {
+        grpc {
+          option 'nano=true'
+        }
+      }
+    }
+    compileClasspath += configurations.testNanoCompile
```
Comment (ejona86) [inline compiler/build.gradle]: It seems this should be unnecessary. Is it?

**Comment by ejona86** [comment]
@zhangkun83 LGTM

File: compiler/build.gradle (hunk 2 above)
Comment (zhangkun83) [inline compiler/build.gradle]: Yes. Removed it.

**Pull request merged**